server/src/main/java/com/thoughtworks/go/config/update/UpdatePackageConfigCommand.java

```diff
@@ -68,6 +75,17 @@ public boolean canContinue(CruiseConfig cruiseConfig) {
         return isAuthorized() && isRepositoryPresent(cruiseConfig) && isIdSame() && isRequestFresh();
     }
 
+    private void updatePackageConfigurationOnAssociatedPipelines(CruiseConfig modifiedConfig) {
+        List<PipelineConfig> pipelinesWithPackage = modifiedConfig.pipelinesAssociatedWithPackage(newPackage);
+        pipelinesWithPackage.forEach(pipelineConfig -> {
+            pipelineConfig.packageMaterialConfigs().forEach(packageMaterialConfig -> {
+                if (packageMaterialConfig.getPackageId().equals(newPackage.getId())) {
```

Comment:
The comparison check has been performed in the `modifiedConfig.pipelinesAssociatedWithPackage` method. Do we require it again.

Reply:
I was thinking of a scenario where a PipleineConfig has more than one package material. We would need this check in those cases right?

Reply:
Yes right - we would need this check then.